### PR TITLE
Refs #126. Adds docker.local socket configuration strategy.

### PR DESCRIPTION
**Pull request by rnorth**

### CODE REVIEW

File: core/src/main/java/org/testcontainers/DockerClientFactory.java [hunk 1]
```diff
@@ -40,7 +36,8 @@
     private static final List<DockerConfigurationStrategy> CONFIGURATION_STRATEGIES =
             asList(new EnvironmentAndSystemPropertyConfigurationStrategy(),
                     new DockerMachineConfigurationStrategy(),
-                    new UnixSocketConfigurationStrategy());
+                    new UnixSocketConfigurationStrategy(),
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java]: Please could you switch the order around a bit? I think the least-surprise sequence should be:

- environment and system properties
- Docker Local (for 'native' support)
- Docker Machine
- Unix socket

Comment (outofcoffee) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java]: got it :)

**Comment by outofcoffee** [comment]
OK, so my changes in the this branch seem to work well on the new Docker for Mac beta:

* **Client:** Docker version 1.11.0, build 4dc5990)
* **Mac app:** Version 1.11.0-beta8 (build: 6072) 3c1bfeb0e86a9403f82302edfea4c4987cc2cb32

Need to check using Docker for Windows beta as well.

**Comment by rnorth** [comment]
I'll give this a try later (beta6) and likely merge for inclusion in 1.0.5.

I'll need to test on Windows (ASAP!) but as we're calling Windows support alpha-grade right now this may not happen before the 1.0.5 release.

File: core/src/main/java/org/testcontainers/DockerClientFactory.java [hunk 2]
```diff
@@ -39,6 +35,7 @@
 
     private static final List<DockerConfigurationStrategy> CONFIGURATION_STRATEGIES =
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java]: Can we make it extendable? IMO ConfigurationStrategy is a good enough abstraction to make it pluggable, i.e. I see some use cases, plus an ability to work around unsupported new stuff like Docker4Mac, or maybe some brand new way of running Docker?

Comment (outofcoffee) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java]: @rnorth - don't - beta9 literially just got released and they're killing `docker.local` and making it `localhost` ;-)

docker_local

**Comment by outofcoffee** [comment]
OK, I've replaced concrete subclasses with configurable socket strategy in https://github.com/testcontainers/testcontainers-java/pull/142/commits/a9bc2085049db8ba925dca7b0415ab25babc6197

Hopefully when the beta stabilises it'll be less like whack-a-mole.

**Comment by rnorth** [comment]
Thanks - let's leave this for a little while and see how things pan out!

**Comment by outofcoffee** [comment]
## Current status
* Mac: Works on Docker for Mac <= beta8. Broken on beta9
* Windows: Works on Docker for Windows beta9.

## Make your voice heard
Please add your support to:
* https://forums.docker.com/t/docker-http-s-api-no-longer-available-in-beta9/10474
* https://forums.docker.com/t/docker4max-compatibility-issues-with-java-client-libraries/8920/2

**Comment by outofcoffee** [comment]
Suggest we supercede this with https://github.com/outofcoffee/testcontainers-java/tree/feature/docker-java-3-socket-proxy-windows-support

**Pull request closed without merging**